transformer_engine/pytorch/attention.py

```diff
@@ -7162,12 +7201,18 @@ def forward(
         # Query, Key, and Value
         # ======================
 
+        fp8_mha = (
+            FP8GlobalStateManager.is_fp8_enabled()
+            and FP8GlobalStateManager.get_fp8_recipe().fp8_mha
+        )
```

Comment:
@timmoon10  Can you review this change?
For now, in `MultiheadAttention`, we want `LayerNormLinear` or `Linear` to produce FP8 output only when `fp8_mha=True and rotary_pos_emb is None`. `MultiheadAttention` is not a `TransformerEngineBaseModule` and doesn't hold the fp8 recipe, so we have to get it from the `FP8GlobalStateManager`.  
Previously I tried to get it from something like `self.layernorm_qkv.fp8_meta['recipe'].fp8_mha`, which doesn't work because `self.layernorm_qkv.fp8_meta` is not initialized until the first fwd step.